research/simulator.py

```
@@ -1,84 +1,72 @@
 
 class Simulator:
   def __init__(self, instance: wfinstances.Instance, logging: bool = True):
     self.instance: wfinstances.Instance = instance
     self.workflow: common.Workflow = instance.workflow
     self.logging = logging
 
-    self.tasks: list[Task] = []
-    self.resources: list[Resource] = []
+    self.tasks: dict[str, Task] = {}
+    self.start_task: Task = None
+    self.resources: dict[str, Resource] = {}
```

Ruff (0.15.12)

[error] Undefined name `Resource`

(F821)